src/lazy/encoder/text/v1_1/value_writer.rs

```diff
@@ -87,13 +88,13 @@ impl<'value, W: Write + 'value> ValueWriter for TextValueWriter_1_1<'value, W> {
         })
     }
 
-    fn eexp_writer<'a>(self, macro_id: impl Into<MacroIdRef<'a>>) -> IonResult<Self::EExpWriter> {
-        let id = macro_id.into();
+    fn eexp_writer<'a>(self, macro_id: impl MacroIdLike<'a>) -> IonResult<Self::EExpWriter> {
+        let id = macro_id.prefer_name();
         let opening_text = match id {
-            MacroIdRef::LocalName(name) => format!("(:{} ", name),
-            MacroIdRef::LocalAddress(address) => format!("(:{} ", address),
+            MacroIdRef::LocalName(name) => format_compact!("(:{} ", name),
+            MacroIdRef::LocalAddress(address) => format_compact!("(:{} ", address),
             MacroIdRef::SystemAddress(system_address) => {
-                format!("(:$ion::{} ", system_address.as_usize())
+                format_compact!("(:$ion::{} ", system_address.as_usize())
```

Comment:
🪧 `format_compact!` produces a `CompactString`, which doesn't require a heap allocation if the resulting text is 24 or fewer bytes long.